test/e2e/grpc_test.go

```diff
@@ -0,0 +1,200 @@
+	deploymentName := names.Revision + "-deployment"
+
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		logger,
+		domain,
+		pkgTest.Retrying(pkgTest.MatchesAny, http.StatusNotFound),
+		"gRPCPingReadyToServe",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint for Route %s at domain %s didn't return success: %v", names.Route, domain, err)
+	}
+
+	host := &domain
+	if !test.ServingFlags.ResolvableDomain {
+		host, err = spoof.GetServiceEndpoint(clients.KubeClient.Kube)
+		if err != nil {
+			t.Fatalf("Could not get service endpoint: %v", err)
+		}
+	}
+
+	logger.Infof("Connecting to grpc-ping using host %q and authority %q", *host, domain)
+
+	waitForScaleToZero := func() {
+		logger.Infof("Waiting for scale to zero")
+		err = pkgTest.WaitForDeploymentState(
+			clients.KubeClient,
+			deploymentName,
+			func(d *v1beta1.Deployment) (bool, error) {
+				logger.Infof("Deployment %q has %d replicas", deploymentName, d.Status.ReadyReplicas)
+				return d.Status.ReadyReplicas == 0, nil
+			},
+			"DeploymentIsScaledDown",
+			test.ServingNamespace,
+			2*time.Minute,
+		)
+		if err != nil {
+			t.Fatalf("Could not scale to zero: %v", err)
+		}
+	}
+
+	conn, err := grpc.Dial(
+		*host+":80",
+		grpc.WithAuthority(domain),
+		grpc.WithInsecure(),
+	)
+	if err != nil {
+		t.Fatalf("fail to dial: %v", err)
+	}
+	defer conn.Close()
+
+	pc := ping.NewPingServiceClient(conn)
+
+	unaryTest := func(t *testing.T) {
+		logger.Info("Testing unary Ping")
+
+		resp, err := pc.Ping(context.TODO(), &ping.Request{Msg: "Hello!"})
+		if err != nil {
+			t.Fatalf("Couldn't send request: %v", err)
+		}
+
+		if resp.Msg != "Hello!" {
+			t.Errorf("Unexpected response. Want %q, got %q", "Hello!", resp.Msg)
+		}
+	}
+
+	streamTest := func(t *testing.T) {
+		logger.Info("Testing streaming Ping")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		stream, err := pc.PingStream(ctx)
+		if err != nil {
+			t.Fatalf("Error creating stream: %v", err)
+		}
+
+		count := 3
+		for i := 0; i < count; i++ {
+			logger.Infof("Sending stream %d of %d", i+1, count)
+
+			payload := make([]byte, 4096)
+			_, err = rand.Read(payload)
+			if err != nil {
+				t.Fatalf("Error generating payload: %v", err)
+			}
+			want := string(payload)
+
+			err = stream.Send(&ping.Request{Msg: want})
+			if err != nil {
+				t.Fatalf("Error sending request: %v", err)
+			}
+
+			resp, err := stream.Recv()
+			if err != nil {
+				t.Fatalf("Error receiving response: %v", err)
+			}
+
+			got := resp.Msg
+
+			if want != got {
+				t.Errorf("Unexpected response. Want %q, got %q", want, got)
+			}
+		}
+
+		stream.CloseSend()
+
+		_, err = stream.Recv()
+		if err != io.EOF {
+			t.Errorf("Expected EOF, got %v", err)
+		}
+	}
+
+	t.Run("unary ping", unaryTest)
+	t.Run("streaming ping", streamTest)
+
+	waitForScaleToZero()
+
+	t.Run("unary ping after scale-to-zero", unaryTest)
+
+	waitForScaleToZero()
```

Comment:
Would commenting this out with `// TODO(#1234): Fix streaming cold start` get this into a start we could check in?

Reply:
Yup, will do!